File: tests/fixtures/openai-realtime/openai-realtime-transcription-model-choice-broken/02-agent-config.ts
export const REALTIME_WS_URL = 'wss://api.openai.com/v1/realtime?model=gpt-4o-realtime-preview';

function buildAgentConfigMsg(agentPrompt: string) {
  return {
    type: 'session.update',
    session: {
      modalities: ['text', 'audio'],
      instructions: agentPrompt,
      input_audio_transcription: {
        model: 'whisper-1',
      },
      turn_detection: { type: 'server_vad' },
    },
  };
}

export default buildAgentConfigMsg;

File: tests/fixtures/openai-realtime/openai-realtime-transcription-model-choice-broken/02_agent_config.py
import json


def build_agent_session_update():
    return json.dumps(
        {
            "type": "session.update",
            "session": {
                "audio": {
                    "input": {
                        "transcription": {"model": "whisper-1"},
                    },
                },
            },
        }
    )
